### PR TITLE
Put up a workaround to accelerate mandlebrot.js on ia32 and x64

### PR DESCRIPTION
**Pull request by chuan9**
Add SIMD.Bool32x4.allTrue in crankshaft, use SIMD.Int32x4 to represent result of SIMD.Float32x4.lessThanOrEqual to accelerate mandlebrot, wirte testcase for SIMID.Bool32x4.anyTrue and SIMD.Bool32x4.allTrue. It's a workaround, the acceleration is as bellow:
Mandelbrot:            Iterations(    715459), SIMD(  300ms), Non-SIMD( 1030ms), Speedup(3.433)

BUG=XWALK-6040

### CODE REVIEW

File: base/trace_event/common [hunk 1]
```diff
@@ -0,0 +1 @@
+Subproject commit e40c41030f44cbd5b6f54081436620f43c3bb08a
```
Comment (rakuco) [inline base/trace_event/common]: Are you sure this one is necessary?

Comment (chuan9) [inline base/trace_event/common]: eh, it's not necessary, I have no idea why it's here, I will remove it and resubmit.

**Comment by chuan9** [comment]
@huningxin Please have a review.

**Comment by huningxin** [comment]
Please update or add corresponding test cases. Thanks!

**Comment by chuan9** [comment]
@huningxin This commit is to accelerate mandlebrot, what kind of test case do you mean? Please give me a hint, thanks

**Comment by huningxin** [comment]
I understand you add `SIMD.Bool32x4.allTrue` and `SIMD.Bool32x4.anyTrue` two new APIs. And it also leads some other API updates. The new APIs and updated APIs should be tested. And for commit log, please specify the what's the new APIs and the implementation (using integer32). I understand it is not only for Mandelbrot benchmark as a workaround, is it?

**Comment by chuan9** [comment]
@huningxin 
For the first commit , I correct implementation SIMD.Bool32x4.contructor and SIMD.Int32x4.select's parameter.
For the second commit, I add SIMD.Bool32x4.allTrue and SIMD.Bool32x4.anyTrue in crankshaft to accelerate mandlebrot.js. These two apis are new to crankshaft not to runtime. And there have some tests in upstream's code(test/simdjs/data/src/ecmascript_simd_tests.js). Should we write tests for these two new crankshaft api(e.g display the acceleration)?

**Comment by huningxin** [comment]
You can force the test function to be optimized by Crankshaft with `%OptimizeFunctionOnNextCall` builtin. And you need also run the mandelbort benchmark to show the difference.

**Comment by chuan9** [comment]
@huningxin Thanks, I added test case for SIMD.Bool32x4.anyTrue and SIMD.Bool32x4.allTrue in bool32x4.js. I have run the mandelbrot benchmark locally and paste it's result. QA will verify it later formally.

**Comment by chuan9** [comment]
@huningxin Please have a review.Thanks

**Comment by chuan9** [comment]
@huningxin Will you help review this patch? Thanks

File: test/mjsunit/harmony/simd/bool32x4.js [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  assertEquals(true, a);
+
+  var b = SIMD.Bool32x4.anyTrue(SIMD.Bool32x4(false, false, false, false));
+  assertEquals(false, b);
+}
+
+testAnyTrue();
+testAnyTrue();
+%OptimizeFunctionOnNextCall(testAnyTrue);
+
```
Comment (huningxin) [inline test/mjsunit/harmony/simd/bool32x4.js]: Please remove this blank line.

File: test/mjsunit/harmony/simd/bool32x4.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+testAnyTrue();
+function testAllTrue() {
```
Comment (huningxin) [inline test/mjsunit/harmony/simd/bool32x4.js]: Please add one blank line between `testAnyTrue()` and `function testAllTrue() {`.

**Comment by huningxin** [comment]
LGTM with nits. Thanks for the patch!

**Comment by rakuco** [comment]
Is there an associated bug report that needs to be referenced here?

**Comment by chuan9** [comment]
@rakuco Yes, It's to fix xwalk-6040. Have done all the reviews, please have a look, thanks

**Comment by chuan9** [comment]
@rakuco Please have a look, It's ok now. Thanks

**Comment by rakuco** [comment]
@chuan9: don't forget to:
- Roll `DEPS.xwalk` in Crosswalk
- Send a PR for the next branch if it's needed there too.

**Comment by chuan9** [comment]
@rakuco There have some missing PRs on next branch compared with master now, may you or some one cherried these from master to next branch?

**Comment by rakuco** [comment]
@mrunalk : ^

**Comment by mrunalk** [comment]
@chuan9 @rakuco I have merged the other SIMD commits into main SIMD commit so all the changes from master except these two commits should be present in next. Have you tried applying your changes on next in your local environment? I think these changes should get applied without any problems. If there is any problem let me know.

**Comment by rakuco** [comment]
@chuan9: the ball is on your court again

**Comment by rakuco** [comment]
Actually I just did it myself now.

**Comment by chuan9** [comment]
@rakuco Thanks!

**Pull request merged**